R-devel is stricter with format strings

```
Result: WARN 
  Found the following significant warnings:
    altrep-rle.c:63:11: warning: format specifies type 'int' but the argument has type 'R_xlen_t' (aka 'long') [-Wformat]
  See ‘/home/hornik/tmp/R.check/r-devel-clang/Work/PKGS/vctrs.Rcheck/00install.out’ for details.
  * used C compiler: ‘Debian clang version 17.0.5 (1)’
```

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,5 +1,7 @@
 # vctrs (development version)
 
+* Internal changes requested by CRAN around C level format strings (#1896).
+
 * Fixed tests related to changes to `dim<-()` in R-devel (#1889).
 
 # vctrs 0.6.4

--- a/src/altrep-rle.c
+++ b/src/altrep-rle.c
@@ -59,7 +59,7 @@ Rboolean altrep_rle_Inspect(SEXP x,
                             int deep,
                             int pvec,
                             void (*inspect_subtree)(SEXP, int, int, int)) {
-  Rprintf("vctrs_altrep_rle (len=%d, materialized=%s)\n",
+  Rprintf("vctrs_altrep_rle (len=%" R_PRIdXLEN_T ", materialized=%s)\n",
           altrep_rle_Length(x),
           R_altrep_data2(x) != R_NilValue ? "T" : "F");
   return TRUE;

--- a/src/vctrs-core.h
+++ b/src/vctrs-core.h
@@ -108,5 +108,15 @@ enum vctrs_dbl dbl_classify(double x);
 
 #define VECTOR_PTR_RO(x) ((const SEXP*) DATAPTR_RO(x))
 
+// Likely supplied in R 4.4.0
+// https://github.com/wch/r-source/commit/38403c9c347dd5426da6009573b087188ec6be04
+#ifndef R_PRIdXLEN_T
+# ifdef LONG_VECTOR_SUPPORT
+#  define R_PRIdXLEN_T "td"
+# else
+#  define R_PRIdXLEN_T "d"
+# endif
+#endif
+
 
 #endif

